### PR TITLE
BUG: is_utc(dateutil_utc)

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/_libs/tslibs/timezones.pyx [hunk 1]
```diff
@@ -30,11 +30,18 @@ from pandas._libs.tslibs.util cimport get_nat, is_integer_object
 cdef int64_t NPY_NAT = get_nat()
 cdef tzinfo utc_stdlib = timezone.utc
 cdef tzinfo utc_pytz = UTC
+cdef tzinfo utc_dateutil1 = dateutil_gettz("UTC")  # NB: *not* the same as tzutc()
```
Comment (mroeschke) [inline pandas/_libs/tslibs/timezones.pyx:33]: Maybe `utc_dateutil_str`

File: pandas/conftest.py [hunk 2]
```diff
@@ -1023,6 +1023,9 @@ def utc_fixture(request):
     return request.param
 
 
+utc_fixture2 = utc_fixture
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/conftest.py:1026]: good catch, wont be needed until both this and the tz_compare PR are in

**Comment by jreback** [comment]
can you also add a whatsnew note

**Comment by jbrockmendel** [comment]
> can you also add a whatsnew note

updated+green

File: pandas/conftest.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jreback) [reply pandas/conftest.py:1026]: remove this?

File: pandas/_libs/tslibs/timezones.pyx (hunk 1 above)
Comment (jreback) [reply pandas/_libs/tslibs/timezones.pyx:33]: change?

File: pandas/conftest.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/conftest.py:1026]: added the extra test below since the other PR got merged

File: pandas/_libs/tslibs/timezones.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/timezones.pyx]: updated + greenish

**Comment by jreback** [comment]
odd why the 32-bit build failed: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=52841&view=logs&j=f39ddbca-85f6-5283-3bfe-05832bcf5208&t=60a5c9af-5075-529a-e870-f4d886d8e7ce

**Comment by jbrockmendel** [comment]
> odd why the 32-bit build failed

 yah ive been seeing that occasionally recently, havent figured out whats going on there

rebased + green

**Review by mroeschke — APPROVED** [review]

**Pull request merged**